extensions/amp-access/0.1/amp-access.js

```diff
@@ -814,10 +817,11 @@ export class AccessService {
    *
    * @param {string} url
    * @param {string} eventLabel A label used for the analytics event for this action
+   * @param {boolean} noReturnUrl Don't perform a RETURN_URL replacement
    * @return {!Promise}
    */
-  loginWithUrl(url, eventLabel = '') {
-    return this.login_(url, eventLabel);
+  loginWithUrl(url, eventLabel = '', noReturnUrl) {
```

Comment:
Sure! So going back a bit to the details of how our backend service works:
* we first send a request to our authorization endpoint with the `CANONICAL_URL` and `READER_ID` (and now, `RETURN_URL` too)
* our authorization endpoint returns the "login urls" which in our case are essentially our purchase URLs for the different options (single article, time passes, subscriptions, etc). Here's an example of one from our development environment:
```
https://web.dev.laterpaytest.net/dialog/add?
  use_dialog_api=False&
  title=Lorem+Ipsum&
  cp=laterpay&
  pricing=EUR145&
  url=http%3A%2F%2Flocalhost%3A8081%2Fexamples%2Farticle-access-laterpay.html&
  ts=1495534548&
  return_url=http%3A%2F%2Flocalhost%3A8081%2Fextensions%2Famp-access%2F0.1%2Famp-login-done.html%3Furl%3Dhttp%3A%2F%2Flocalhost%3A8081%2Fexamples%2Farticle-access-laterpay.amp.html%23success%3Dtrue&
  article_id=premiumcontent%3A9aa68278-173a-48ec-be04-6b135654bfa9&
  muid=amp-Irryr4n4MTYt8tgAClLDre8_g_LzB5_7nTx9Q8sWUPSeuoAb-rJ6UkMfuk_I6kI0&
  hmac=5243e6e9f6bec6b08151c7d3efe37f7844159e2d840e4b435b2281b6
```
* Those urls already have the return URL baked in. That's why we need to send it on our initial authorization request.
* Because those urls are signed, if anything is changed in them, they won't be recognized as a valid URL by our payment dialog system

This is why we need to make sure that additional stuff isn't being added to the URL after we get it back from the authorization endpoint.

Does that make sense?

Also this was my suggestion of how to handle this particular issue but we're definitely open to suggestions of a better way of handling this problem.